examples/container-service-extension-4.0/install/step1/3.9-cse-4.0-install-step1.tf

```diff
@@ -0,0 +1,111 @@
+#   before applying this configuration.
+#
+# * The installation process is split into two steps as Providers will need to generate an API token for the created
+#   CSE administrator user, in order to use it with the CSE Server that will be deployed in the second step.
+#
+# * This step will only create the required Runtime Defined Entity (RDE) Interfaces, Types, Role and finally
+#   the CSE administrator user.
+#
+# * Rename "terraform.tfvars.example" to "terraform.tfvars" and adapt the values to your needs.
+#   Other than that, this snippet should be applied as it is.
+#   You can check the comments on each resource/data source for more help and context.
+# ------------------------------------------------------------------------------------------------------------
+
+# VCD Provider configuration. It must be at least v3.9.0 and configured with a System administrator account.
+terraform {
+  required_providers {
+    vcd = {
+      source  = "vmware/vcd"
+      version = ">= 3.9"
+    }
+  }
+}
+
+provider "vcd" {
+  url                  = "${var.vcd_url}/api"
+  user                 = var.administrator_user
+  password             = var.administrator_password
+  auth_type            = "integrated"
+  sysorg               = var.administrator_org
+  org                  = var.administrator_org
+  allow_unverified_ssl = var.insecure_login
+  logging              = true
+  logging_file         = "cse_install_step1.log"
+}
+
+# This is the interface required to create the "VCDKEConfig" Runtime Defined Entity Type.
+resource "vcd_rde_interface" "vcdkeconfig_interface" {
+  vendor  = "vmware"
+  nss     = "VCDKEConfig"
+  version = "1.0.0"
+  name    = "VCDKEConfig"
+}
+
+# This resource will manage the "VCDKEConfig" RDE Type required to instantiate the CSE Server configuration.
+# The schema URL points to the JSON schema hosted in the terraform-provider-vcd repository.
+resource "vcd_rde_type" "vcdkeconfig_type" {
+  vendor        = "vmware"
+  nss           = "VCDKEConfig"
+  version       = "1.0.0"
+  name          = "VCD-KE RDE Schema"
+  schema_url    = "https://raw.githubusercontent.com/vmware/terraform-provider-vcd/main/examples/container-service-extension-4.0/schemas/vcdkeconfig-type-schema.json"
+  interface_ids = [vcd_rde_interface.vcdkeconfig_interface.id]
+}
+
+# This RDE Interface exists in VCD, so it must be fetched with a RDE Interface data source. This RDE Interface is used to be
+# able to create the "capvcdCluster" RDE Type.
+data "vcd_rde_interface" "kubernetes_interface" {
+  vendor  = "vmware"
+  nss     = "k8s"
+  version = "1.0.0"
+}
+
+# This RDE Interface will create the "capvcdCluster" RDE Type required to create Kubernetes clusters.
+# The schema URL points to the JSON schema hosted in the terraform-provider-vcd repository.
+resource "vcd_rde_type" "capvcdcluster_type" {
+  vendor        = "vmware"
+  nss           = "capvcdCluster"
+  version       = var.capvcd_version
```

Comment:
CAPVCD Product version can be different from RDE version.
In this particular case, it is a coincidence that both happened to be 1.1.0

See the below history
CAPVCD 0.5.0 - RDE 1.0.0
CAPVCD 1.0.0 - RDE 1.1.0
CAPVCD 1.1.0 - RDE 1.1.0

Reply:
Aah, I get it now!